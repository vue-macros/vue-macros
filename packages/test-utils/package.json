{
  "name": "@vue-macros/test-utils",
  "version": "3.0.0-beta.5",
  "description": "Test utilities for Vue Macros.",
  "type": "module",
  "keywords": [
    "vue-macros",
    "macros",
    "vue",
    "sfc",
    "setup",
    "script-setup",
    "test-utils"
  ],
  "license": "MIT",
  "homepage": "https://vue-macros.dev",
  "bugs": {
    "url": "https://github.com/vue-macros/vue-macros/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/vue-macros/vue-macros.git",
    "directory": "packages/test-utils"
  },
  "author": "三咲智子 Kevin Deng <sxzz@sxzz.moe>",
  "funding": "https://github.com/sponsors/vue-macros",
  "files": [
    "dist"
  ],
  "main": "dist/index.js",
  "module": "dist/index.js",
  "exports": {
    ".": {
      "dev": "./src/index.ts",
      "default": "./dist/index.js"
    },
    "./*": "./*"
  },
  "publishConfig": {
    "access": "public",
    "exports": {
      ".": "./dist/index.js",
      "./*": "./*"
    },
    "tag": "next"
  },
  "scripts": {
    "build": "tsup",
    "dev": "DEV=true tsup --watch"
  },
  "dependencies": {
    "@rollup/plugin-json": "catalog:",
    "@rollup/plugin-node-resolve": "catalog:",
    "@sxzz/test-utils": "catalog:",
    "@vitejs/plugin-vue-jsx": "catalog:",
    "rollup": "catalog:",
    "unplugin-oxc": "catalog:",
    "unplugin-vue": "catalog:"
  },
  "devDependencies": {
    "vite": "catalog:",
    "vue": "catalog:"
  },
  "engines": {
    "node": ">=20.18.0"
  }
}
