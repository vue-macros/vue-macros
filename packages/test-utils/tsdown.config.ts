import { defineConfig } from 'tsdown/config'

export default defineConfig({
  entry: ['src/index.ts'],
})
