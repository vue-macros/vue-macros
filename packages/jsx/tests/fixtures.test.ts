import { resolve } from 'node:path'
import { rollupBuild, RollupVueJsx, testFixtures } from '@vue-macros/test-utils'
import { describe } from 'vitest'
import VueJsxMacros from '../src/rollup'

describe('fixtures', async () => {
  await testFixtures(
    ['tests/fixtures/**/*'],
    (_, id) => {
      return rollupBuild(id, [VueJsxMacros(), RollupVueJsx()])
    },
    {
      cwd: resolve(__dirname, '..'),
      promise: true,
    },
  )
})
