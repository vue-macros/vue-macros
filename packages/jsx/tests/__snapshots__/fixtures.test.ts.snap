// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixtures > tests/fixtures/index.tsx 1`] = `
"// index.js
import { createVNode, Fragment } from 'vue';

var index = () => {
  let foo = 1;
  return createVNode(Fragment, null, [foo === 1 ? createVNode("div", null, [foo]) : null]);
};

export { index as default };
"
`;
